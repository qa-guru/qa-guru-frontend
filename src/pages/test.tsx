import { FC } from "react";
import { useParams } from "react-router-dom";

import TestContainer from "features/test/containers/test-container";

const TestPage: FC = () => {
  const { testId, trainingId, lectureId } = useParams<{
    testId: string;
    trainingId: string;
    lectureId: string;
  }>();

  // Проверяем, что все параметры присутствуют
  if (!testId || !trainingId || !lectureId) {
    return <div>Ошибка: отсутствуют необходимые параметры</div>;
  }

  return (
    <TestContainer
      testId={testId}
      trainingId={trainingId}
      lectureId={lectureId}
    />
  );
};

export default TestPage;
