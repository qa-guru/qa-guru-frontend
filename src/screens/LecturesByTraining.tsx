import React from "react";
import { Button, Typography } from "@mui/material";
import { useNavigate } from "react-router-dom";
import ArrowBackIcon from "@mui/icons-material/ArrowBack";
import TrainingItemTitle from "../features/Training/TrainingItemTitle";
import TrainingLectures from "../features/Lecture/TrainingLectures";
import { primary } from "../theme/colors";

const style = {
  button: {
    mb: "25px",
    color: primary.main,
  },
  icon: { mr: "10px" },
};

const LecturesByTraining: React.FC = () => {
  const navigate = useNavigate();

  return (
    <>
      <Button
        variant="outlined"
        sx={style.button}
        onClick={() => navigate("/")}
      >
        <ArrowBackIcon sx={style.icon} />
        <Typography textTransform="none" variant="subtitle1">
          К списку курсов
        </Typography>
      </Button>
      <TrainingItemTitle />
      <TrainingLectures />
    </>
  );
};

export default LecturesByTraining;
