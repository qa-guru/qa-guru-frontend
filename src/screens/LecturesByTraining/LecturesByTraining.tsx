import React from "react";
import { Button, Typography } from "@mui/material";
import { useNavigate } from "react-router-dom";
import ArrowBackIcon from "@mui/icons-material/ArrowBack";
import TrainingItemTitle from "../../features/Training/TrainingItemTitle";
import TrainingLectures from "../../features/Lecture/TrainingLectures";

const style = {
  button: {
    mb: "25px",
  },
  icon: { mr: "10px" },
};

const LecturesByTraining: React.FC = () => {
  let navigate = useNavigate();

  return (
    <>
      <Button sx={style.button} onClick={() => navigate("/")}>
        <ArrowBackIcon sx={style.icon} />
        <Typography textTransform="none" variant="subtitle1">
          К списку курсов
        </Typography>
      </Button>
      <TrainingItemTitle />
      <TrainingLectures />
    </>
  );
};

export default LecturesByTraining;
