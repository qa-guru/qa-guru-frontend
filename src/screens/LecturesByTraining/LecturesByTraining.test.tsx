import { MemoryRouter } from "react-router-dom";
import { render } from "../../test/utilsTest";
import LecturesByTraining from "./LecturesByTraining";

describe("LecturesByTraining", () => {
  it("the component is render", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <LecturesByTraining />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
