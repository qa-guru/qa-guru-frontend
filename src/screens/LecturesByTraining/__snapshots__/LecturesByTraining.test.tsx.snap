// Vitest Snapshot v1

exports[`LecturesByTraining > the component is render 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1b30wvl-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1pdpm1l-MuiSvgIcon-root"
      data-testid="ArrowBackIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
      />
    </svg>
    <h6
      class="MuiTypography-root MuiTypography-subtitle1 css-13zisa9-MuiTypography-root"
    >
      К списку курсов
    </h6>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root css-1ji7omv-MuiBackdrop-root"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
      role="progressbar"
      style="width: 100px; height: 100px;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
  </div>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root css-1ji7omv-MuiBackdrop-root"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
      role="progressbar"
      style="width: 100px; height: 100px;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
  </div>
</DocumentFragment>
`;
